### PR TITLE
Generate Sentry BOM

### LINKED ISSUE #1484: Support Sentry BOM
https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#Importing_Dependencies
https://docs.gradle.org/current/userguide/platforms.html#sub:bom_import

### PR DESCRIPTION
**Pull request by maciejwalkowiak**
## :scroll: Description

Generate Sentry BOM

## :bulb: Motivation and Context

Fixes #1484

## :pencil: Checklist

- [x] I reviewed the submitted code
- [ ] I added tests to verify the changes
- [ ] I updated the docs if needed
- [ ] No breaking changes

## :crystal_ball: Next steps

Update `release.kt` script BOM to publish to Maven central.

### CODE REVIEW

**Comment by maciejwalkowiak** [comment]
Since I am not sure how to release the BOM file, the `release.kt` file will have to be updated likely when doing the release to find out if its enough to provide `-Dfile` property, or it has to be in `-DpomFile` etc.

**Comment by marandaneto** [comment]
I published it locally and tried using `mavenLocal` but it does not work

> A problem occurred configuring project ':app'.
> > Could not resolve all dependencies for configuration ':app:debugCompileClasspath'.
>    > Could not find io.sentry:sentry-bom:5.0.0-beta.5-SNAPSHOT.
>      Required by:
>          project :app
>    > Could not find io.sentry:sentry:.
>      Required by:
>          project :app

the other thing is, `release.kts` won't work as it's a single pom and does not contain any javadoc, sources file, so we'd need to special case this project in that script.
the folder `distributions` also does not contain the `sentry-bom` subfolder, I think it's not generating the zip file, this should also be somehow solved.

**Comment by marandaneto** [comment]
just tried locally:

```
    implementation platform("io.sentry:sentry-bom:{version}")
    implementation "io.sentry:sentry-android"
```

and it worked, thanks @maciejwalkowiak

**Comment by marandaneto** [comment]
@maciejwalkowiak for docs https://github.com/getsentry/sentry-docs/issues/3652

**Review by marandaneto — APPROVED** [review]

**Comment by marandaneto** [comment]
@maciejwalkowiak do you prefer opening a new PR for fixing the release script or this one?
it can be tested locally if temp. modifying settings.xml to maven local

**Comment by maciejwalkowiak** [comment]
How the settings should be modified?

**Comment by marandaneto** [comment]
> How the settings should be modified?

ups sorry, `release.kts`, instead of using `repositoryUrl` as the cloud provider, set it to use maven local, so you can double check the deployment in `~/.m2/repository`

**Comment by maciejwalkowiak** [comment]
@marandaneto updated the release script. Works well when I tested it locally.

File: scripts/release.kts [hunk 1]
```diff
@@ -49,29 +49,42 @@ File(path)
...
 
             val file: String
 
             val androidFile = folder
                     .listFiles { it -> it.name.contains("release") && it.extension == "aar" }
                     .firstOrNull()
 
-            if (androidFile != null) {
-                file = androidFile.path
+            val bomFile = folder
```
Comment (marandaneto) [inline scripts/release.kts:60]: @iker-barriocanal these changes will need to be updated in the craft release too, right?

File: CHANGELOG.md [hunk 2]
```diff
@@ -51,6 +51,7 @@ Thank you:
 
 ## 5.0.0-beta.6
 
+* Feat: Generate Sentry BOM (#1486)
```
Comment (marandaneto) [inline CHANGELOG.md:54]: should be under Unreleased

Comment (maciejwalkowiak) [reply CHANGELOG.md]: 👍

**Comment by maciejwalkowiak** [comment]
@marandaneto should we merge it or not yet?

**Comment by marandaneto** [comment]
> @marandaneto should we merge it or not yet?

I will test later but also need to check out https://github.com/getsentry/sentry-java/pull/1486#discussion_r657718422
otherwise, the Draft of Craft will be broken already

File: scripts/release.kts (hunk 1 above)
Comment (iker-barriocanal) [reply scripts/release.kts:60]: Yes, will update.

**Comment by marandaneto** [comment]
@iker-barriocanal I am about to merge this, is craft ready to do a `bom` publish too?

File: scripts/release.kts [hunk 3]
```diff
@@ -49,29 +49,42 @@ File(path)
...
+                .listFiles { it -> it.extension == "jar" }
+                .isEmpty()
+
+            if (bomFile) {
+                val command = "./mvnw gpg:sign-and-deploy-file " +
+                    "-Dfile=$pomFile " +
+                    "-DpomFile=$pomFile " +
+                    "-DrepositoryId=$repositoryId " +
+                    "-Durl=$repositoryUrl " +
+                    "--settings $settingsPath"
```
Comment (marandaneto) [inline scripts/release.kts]: @iker-barriocanal how craft will handle this, as it's a different way of calling maven CLI for a bom file?

**Comment by bruno-garcia** [comment]
An update, I merged the docs which I assume I'll need to revert. But to let y'all know we're just blocked on being able to release this. Hopefully @iker-barriocanal can help us out on this one too

**Comment by iker-barriocanal** [comment]
@bruno-garcia @marandaneto opened a PR to support this in Craft: https://github.com/getsentry/craft/pull/270.

**Comment by bruno-garcia** [comment]
> @bruno-garcia @marandaneto opened a PR to support this in Craft: [getsentry/craft#270](https://github.com/getsentry/craft/pull/270).

And it's merged! thanks Iker, I guess we're close to merging this one then

**Pull request merged**